File: Ahorn/triggers/StartCountdownTrigger.jl
module XaphanHelperStartCountdownTrigger

using ..Ahorn, Maple

@mapdef Trigger "XaphanHelper/StartCountdownTrigger" StartCountdownTrigger(x::Integer, y::Integer, time::Number=60.00, startFlag::String="", activeFlag::String="", shake::Bool=false, explosions::Bool=false, crossChapter::Bool=false, dialogID::String="", messageTimer::Number=5.00, fastMessageDisplay::Bool=false, messageColor::String="FFFFFF")

const placements = Ahorn.PlacementDict(
    "Start Countdown Trigger (Xaphan Helper)" => Ahorn.EntityPlacement(
        StartCountdownTrigger,
        "rectangle"
    )
)

end

File: Loenn/triggers/StartCountdownTrigger.lua
local StartCountdownTrigger = {}

StartCountdownTrigger.name = "XaphanHelper/StartCountdownTrigger"
StartCountdownTrigger.fieldOrder = {
    "x", "y", "width", "height", "time", "startFlag", "activeFlag", "shake", "explosions", "crossChapter", "dialogID", "messageTimer", "fastMessageDisplay", "messageColor"
}
StartCountdownTrigger.fieldInformation = {
    messageColor = {
        fieldType = "color"
    }
}
StartCountdownTrigger.placements = {
    name = "StartCountdownTrigger",
    data = {
        time = 60.00,
        startFlag = "",
        activeFlag = "",
        shake = false,
        explosions= false,
        crossChapter = false,
        dialogID = "",
        messageTimer = 5.00,
        fastMessageDisplay = false,
        messageColor = "FFFFFF"
    }
}

return StartCountdownTrigger